packages/data-context/src/actions/NotificationActions.ts

```diff
@@ -63,37 +63,45 @@ export class NotificationActions {
     this.sendRunCompletedNotification(newRun.runNumber, newRun.status.toLowerCase() as NotifyWhenRunCompletes)
   }
 
-  private showRunNotification (body: string, runNumber: number) {
-    this.ctx.actions.electron.showSystemNotification(this.projectTitle, body, () => this.onNotificationClick(runNumber))
+  async #showRunNotification (body: string, runNumber: number) {
+    try {
+      const cloudProjectMetadata = this.ctx.coreData.cloudProject.metadata ?? await this.ctx.actions.cloudProject.fetchMetadata()
```

Comment:
I couldn't find a single place that makes sense to call this earlier in the lifecycle, so I'm just calling it when we first show a notification. We cache the data. Said data is cleared when we call `clearCloudProject`, which is called in the `clearCurrentProject()` method (where all cleanup between projects occurs).

This does not really feel ideal, but it follows the convention we have (clean up in one place) and at the very least it is **simple** - you don't need to figure out how / if the cloud metadata is set, it's called right before the data is needed.

We definitely need a better pattern around the general "give me the cloud metadata associated with this project" concept, I think we need more time and scenarios to figure out what that is, though.

The only concern here is this will block showing the notification until the request finishes - that said, we are  constantly in sync with the Cloud via the runs polling anyway, so we know the Cloud is alive and responding  - worse case scenario is your notification might be delayed by a second or two, which won't be an issue - it's not really real time anyway right now, since we are polling.